TIBERIANDAWN/LUA_API.CPP

```diff
@@ -6,15 +6,17 @@ static void Load_Api_Functions()
 {
     Log_Debug("Initialising Lua API functions");
 
-    Register_Lua_Function("getGameRule", Lua_Get_Game_Rule);
-    Register_Lua_Function("setGameRule", Lua_Set_Game_Rule);
+    Register_Lua_Function("getStringGameRule", Lua_Get_String_Game_Rule);
```

Comment:
Should we be creating type-specific get/set functions? I'd suggest we have the internal rules structure strongly typed as it is now, and when a function sends a get/set request we do any type conversion within the parser. Asking mod makers to remember what kind of variable is what type *seems* like a reasonable thing to ask for, but in my experience it just causes a lot of bug requests and complaints :D

Reply:
Wait, I just saw the third commit, and it looks like you're already wrapping these and doing this. Sorry for the confusion!

Reply:
No worries - yeah basically these api functions will be `undocumented` and have a wrapper written in lua to make them easy to use. Had to rip out the `RulesIniSection` class because it was causing strange memory corruption issues - will add back in later release